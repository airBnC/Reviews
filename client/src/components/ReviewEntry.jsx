import React from 'react';
import PropTypes from 'prop-types';

const ReviewEntry = props => (
  <div>
    <h6> {props.review.review_date} </h6>
    <h6> {props.review.review_text} </h6>
    <h6> {props.review.first_name} </h6>
  </div>
);

ReviewEntry.propTypes = {
  review: PropTypes.shape({
    review_date: PropTypes.string,
    review_text: PropTypes.string,
    first_name: PropTypes.string,
    last_name: PropTypes.string,
  }),
};

ReviewEntry.defaultProps = {
  review: {
    review_date: '2016-11-04',
    review_text: 'Laborum eius id rerum et in ratione',
    first_name: 'Toby',
    last_name: 'Keith',
  },
};

export default ReviewEntry;
