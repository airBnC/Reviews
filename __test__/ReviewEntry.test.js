import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ReviewEntry from '../client/src/components/ReviewEntry';

configure({ adapter: new Adapter() });

describe('ReviewEntry', () => {
  it('should render correctly', () => {
    const output = shallow(<ReviewEntry />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
