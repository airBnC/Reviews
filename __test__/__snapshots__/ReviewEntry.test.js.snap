// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewEntry should render correctly 1`] = `
<div>
  <h5>
     Begin Review Entry 
  </h5>
  <h6>
     
    2016-11-04
     
  </h6>
  <h6>
     
    Laborum eius id rerum et in ratione
     
  </h6>
  <h6>
     
    Toby
     
  </h6>
</div>
`;
